pyanaconda/image.py

```diff
@@ -172,14 +172,27 @@ def mountImage(isodir, tree):
             break
 
 
-# Return the first Device instance containing valid optical install media
-# for this product.
 def opticalInstallMedia(devicetree):
-    retval = None
+    """Find a device with a valid optical install media.
 
-    # Search for devices identified as cdrom along with any other
-    # device that has an iso9660 filesystem. This will catch USB media
-    # created from ISO images.
+    FIXME: This is a temporary workaround for UI.
+    """
+    return find_optical_install_media(devicetree)
+
+
+def find_optical_install_media(devicetree):
```

Comment:
This should be moved to payload module later. Storage module should onlyprovide list of valid devices and the check if it is an installation media should be done in the Payload module instead.

Reply:
Fixed.